### PR TITLE
Add backing Azure Table data model and client

### PR DESCRIPTION
**Pull request by Arkatufus**
- Add proto file
- Add cluster member Azure data model
- Add Azure table client
- Modify AzDO build script to install Azurite locally

### CODE REVIEW

File: src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        internal TableEntity Raw { get; }
+        internal ClusterMemberProto Proto { get; }
+        
+        public string ServiceName => PartitionKey;
+        public string Host => Proto.Host;
+        public IPAddress Address => IPAddress.Parse(Proto.Address);
+        public int Port => Proto.Port;
+        public DateTime Created => Proto.Created.ToDateTime();
+        public DateTime LastUpdate { get; }
```
Comment (Arkatufus) [inline src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs:49]: As per changes in the spec, the cluster member table data model is stripped to the bare minimum. The only fields exposed are the ones needed by the query for TTL scheme support.
All the rest of the supporting data are packed inside a byte array using protobuf instead, for data versioning reasons

File: build-system/azure-pipeline.template.yaml [hunk 2]
```diff
@@ -26,6 +26,12 @@ jobs:
         inputs:
           version: 3.1.x
       # Linux or macOS
+      - bash: |
+          sudo npm install -g azurite@3.17.1
+          sudo mkdir azurite
+          sudo azurite --silent --location azurite --debug azirute/debug.log &
```
Comment (Aaronontheweb) [inline build-system/azure-pipeline.template.yaml:32]: You wanted to do this over `Docker.NET`?

File: src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        public static TableEntity CreateEntity(
+            string serviceName,
+            IPAddress address,
+            int port)
+        {
+            var now = DateTime.UtcNow;
+            var proto = new ClusterMemberProto
+            {
+                Host = Dns.GetHostName(),
+                Address = address.MapToIPv4().ToString(),
```
Comment (Aaronontheweb) [inline src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs:70]: Why have host and address as separate fields? Can't a node have either one or the other?

File: src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+using Akka.Annotations;
+using Akka.Discovery.Azure.Model;
+using Akka.Event;
+using Azure;
+using Azure.Data.Tables;
+
+namespace Akka.Discovery.Azure
+{
+    [InternalApi]
+    public class ClusterMemberTableClient
```
Comment (Aaronontheweb) [inline src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs:22]: Does this need to be `public` or can we make it `internal`?

File: src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            return list.ToImmutableList();
+        }
+
+        internal async Task<bool> UpdateAsync(CancellationToken token = default)
+        {
+            if (!await EnsureInitializedAsync(token))
+                return false;
+
+            var original = _entity.LastUpdate;
+            _entity = _entity.Update();
```
Comment (Aaronontheweb) [inline src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs:101]: This updates the timestamp?

File: build-system/azure-pipeline.template.yaml (hunk 2 above)
Comment (Arkatufus) [reply build-system/azure-pipeline.template.yaml:32]: No, the Azurite Docker image rarely got updated and it is linux only. Its the same reason why we switched to this way of doing it in Akka.Persistence.Azure.

Comment (Aaronontheweb) [reply build-system/azure-pipeline.template.yaml:32]: Got it, thanks

File: src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs (hunk 3 above)
Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/Model/ClusterMember.cs:70]: Ok, got it.

File: src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs (hunk 5 above)
Comment (Arkatufus) [reply src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs:101]: Yes, the ClusterMember class is immutable, this creates a new ClusterMember instance with an updated LastUpdate field.

File: src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs (hunk 4 above)
Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs:22]: Don't they all get started internally by the `Akka.Discovery` plugin though? i.e. the user doesn't have to launch any of those components.

File: src/discovery/azure/Akka.Discovery.Azure/Akka.Discovery.Azure.csproj [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    <ItemGroup>
+      <None Remove="protobuf\ClusterMemberProto.proto" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Akka.Cluster" Version="$(AkkaVersion)" />
+      <PackageReference Include="Azure.Data.Tables" Version="12.6.0" />
+      <PackageReference Include="Google.Protobuf" Version="3.21.2" />
+      <PackageReference Include="Grpc.Tools" Version="2.40.0">
```
[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/Akka.Discovery.Azure.csproj:16]: LGTM then

File: src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs (hunk 4 above)
Comment (Arkatufus) [reply src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs:22]: Right, I'll change it.

Comment (Arkatufus) [reply src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs]: Done

**Comment by Aaronontheweb** [comment]
@Arkatufus you might need to rebase on `dev` once https://github.com/akkadotnet/Akka.Management/pull/667 in order to get CI to run for this, FYI.

**Comment by Arkatufus** [comment]
Done

**Comment by Aaronontheweb** [comment]
@Arkatufus nice work - merged.

**Pull request merged**